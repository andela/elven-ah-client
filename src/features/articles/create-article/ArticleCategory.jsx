import React from 'react';
import Select from 'react-select';
import PropTypes from 'prop-types';

import categoryOptions from './categoryOptions';

const ArticleCategory = ({ category, handleChange }) => (
  <Select
    value={category}
    onChange={handleChange}
    options={categoryOptions}
    placeholder="Select a category"
  />
);

ArticleCategory.propTypes = {
  category: PropTypes.shape({}).isRequired,
  handleChange: PropTypes.func.isRequired,
};

export default ArticleCategory;
