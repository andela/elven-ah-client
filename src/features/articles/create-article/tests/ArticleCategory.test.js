import React from 'react';
import { shallow } from 'enzyme';
import Select from 'react-select';
import ArticleCategory from '../ArticleCategory';

describe('Article Category component', () => {
  it('should render without errors', () => {
    const wrapper = shallow(<ArticleCategory
      category={{ value: 1, label: 'Technology' }}
      handleChange={() => jest.fn()}
    />);
    expect(wrapper.find(Select)).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });
});
