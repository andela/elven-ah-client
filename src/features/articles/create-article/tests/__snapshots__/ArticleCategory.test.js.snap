// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article Category component should render without errors 1`] = `
<StateManager
  defaultInputValue=""
  defaultMenuIsOpen={false}
  defaultValue={null}
  onChange={[Function]}
  options={
    Array [
      Object {
        "label": "Politics",
        "value": 1,
      },
      Object {
        "label": "Design",
        "value": 2,
      },
      Object {
        "label": "Africa",
        "value": 3,
      },
      Object {
        "label": "Technology",
        "value": 4,
      },
      Object {
        "label": "Culture",
        "value": 5,
      },
      Object {
        "label": "Business",
        "value": 6,
      },
      Object {
        "label": "Romance",
        "value": 7,
      },
      Object {
        "label": "Health",
        "value": 8,
      },
      Object {
        "label": "Style",
        "value": 9,
      },
      Object {
        "label": "Travel",
        "value": 10,
      },
      Object {
        "label": "DIY",
        "value": 11,
      },
      Object {
        "label": "Opinion",
        "value": 12,
      },
    ]
  }
  placeholder="Select a category"
  value={
    Object {
      "label": "Technology",
      "value": 1,
    }
  }
/>
`;
